packages/style-engine/class-wp-style-engine.php

```diff
@@ -165,6 +184,11 @@ protected static function get_slug_from_preset_value( $style_value, $property_ke
 	 */
 	protected static function get_classnames( $style_value, $style_definition ) {
 		$classnames = array();
+
+		if ( empty( $style_value ) ) {
```

Comment:
Minor question: Is it worth future-proofing this a little in case the classname generation extends beyond presets in future? 

This check would still return a false positive for `'0'`. I'm thinking of things like the Cover blocks opacity classes like `has-background-dim-0`.

Reply:
That's an interesting case, thanks for raising it. 

There's no requirement for such values here yet, but if we hit the cover block one day it'll be there.

I was just trying to prevent the code doing any further work for now. But there's no harm in removing the check since it wasn't there before. 👍

Reply:
```suggestion
		if ( empty( $style_value ) && '0' !== $style_value ) {
```

I was more suggesting that we tweak that check to allow `'0'` values rather than remove it. Sorry for the confusion.

Reply:
All good. I read the comment wrong. In that case it's probably safe to leave as it was until we get to individual blocks in 2025 😄 

The fact that the style engine generates classes at all is still a bit contentious, that is, whether it should just stick to CSS.